xds/googledirectpath/googlec2p_test.go

```diff
@@ -251,3 +252,15 @@ func TestBuildXDS(t *testing.T) {
 		})
 	}
 }
+
+// TestBuildFailsWhenCalledWithAuthority asserts that c2p resolver.build
+// fails when called with a URL with an authority.
+func TestBuildFailsWhenCalledWithAuthority(t *testing.T) {
+	builder := resolver.Get(c2pScheme)
+	targetURL, _ := url.Parse("google-c2p://an-authority/resource")
+
+	_, err := builder.Build(resolver.Target{URL: *targetURL}, nil, resolver.BuildOptions{})
+	if err == nil {
+		t.Fatalf("c2p resolver should not support target's with authorities, but build succeeded")
+	}
```

Comment:
How about replacing this with:

```golang
uri := "google-c2p://an-authority/resource"
cc, err := grpc.Dial(uri, grpc.WithTransportCredentials(insecure.NewCredentials())
defer func() {
  if cc != nil {
    cc.Close()
  }
}
wantErr := "google-c2p URI scheme does not support authorities"
if err == nil || !strings.Contains(err.Error(), wantErr) {
  cc.Close()
  t.Fatalf("grpc.Dial(%s) returned error: %v, want: %v", uri, err, wantErr) 
}
```

This is a little longer than what you have, but the good thing is that we can ensure that the underlying error is propagated all the way up to the user.

Reply:
sgtm. ill update it. This is more end to end than the coverage specific unit test i have in there